### PR TITLE
SAMZA-2147: Improve the documentation of the samza release process.

### PR DESCRIPTION
**Pull request by shanthoosh**
Currently the RELEASE.md which outlines the samza release process in open-source has the following problems:

* There is no clear separation between the steps that has to be followed before and after the successful vote completion.
* There are formatting errors in the bash commands that has to be executed to validate the release.

This PR fixes the above problems.

### CODE REVIEW

**Comment by shanthoosh** [comment]
@dxichen Please take a look when you have a chance.

**Comment by shanthoosh** [comment]
@dxichen 
Here is the screen shot of the `RELEASE.md` file rendered after this patch. 
screencapture-file-home-svenkata-Downloads-Samza-Release-Procedure-md-html-2019-03-25-18_03_16

File: RELEASE.md [hunk 1]
```diff
@@ -103,48 +127,54 @@ from reading keyboard input (because we need `org.gradle.jvmargs` set).
...
 
 * Set the desired `scalaSuffix` in `gradle.properties`.
 * Run `./gradlew clean uploadArchives` to generate and upload the Samza artifacts.
-* Login to the [repository web interface](https://repository.apache.org/) with your Apache LDAP 
-credentials, "close" the created repository and "release" it.
+
+# After the VOTE has passed
+
+If the VOTE has successfully passed on the release candidate, you can log in to the  [repository web interface](https://repository.apache.org) (same as above) and "release"  the org.apache.samza repository listed under "Staging Repositories". This may take a minute or so.
+This will publish the samza release artifacts to the open source [maven repository](https://repo1.maven.org/maven2)
```
Comment (dxichen) [inline RELEASE.md]: Add period at end of sentence

File: RELEASE.md [hunk 2]
```diff
@@ -22,78 +22,102 @@ Before you start, here are a few prerequisite steps that would be useful later:
...
     git tag -s release-$VERSION-rc0 -m "Apache Samza $VERSION release candidate 0"
+```
 
 Push the release tag to remote repository:
 
+```bash
     git push origin release-$VERSION-rc0
+```
 
 Edit `$HOME/.gradle/gradle.properties` and add your GPG key information (without the comments):
```
Comment (dxichen) [inline RELEASE.md]: can you add a note to remove all trailing whitespace too?

File: RELEASE.md (hunk 1 above)
Comment (shanthoosh) [reply RELEASE.md]: Done.

File: RELEASE.md (hunk 2 above)
Comment (shanthoosh) [reply RELEASE.md]: Trailing white-space does not matter. The nexus-plugin handles it.

**Pull request merged**